src/media-segment-request.js

```diff
@@ -168,11 +168,14 @@ const handleInitSegmentResponse =
 
   segment.map.bytes = new Uint8Array(request.response);
 
-  if (isLikelyWebmData(segment.map.bytes)) {
+  const type = detectContainerForBytes(segment.map.bytes);
+
+  if (type !== 'mp4') {
     return finishProcessingFn({
       status: request.status,
-      message: 'Found unsupported WebM initialization segment at URL: ' + request.uri,
+      message: `Found unsupported ${type || 'unknown'} container for initialization segment at URL: ${request.uri}`,
       code: REQUEST_ERRORS.FAILURE,
+      internal: true,
```

Comment:
Are we ok with this being an internal error? Seems like a feature we lack rather than an actual error to me, but if we want to warn in the console that could be fine.